### ISSUE TITLE
Go module downloaded with no go files

### ISSUE THREAD

**Issue by emily33901** [golang/go#38065]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.14.1 windows/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
set GO111MODULE=
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\josh\AppData\Local\go-build
set GOENV=C:\Users\josh\AppData\Roaming\go\env
set GOEXE=.exe
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOINSECURE=
set GONOPROXY=
set GONOSUMDB=
set GOOS=windows
set GOPATH=E:\go
set GOPRIVATE=
set GOPROXY=https://proxy.golang.org,direct
set GOROOT=c:\go
set GOSUMDB=sum.golang.org
set GOTMPDIR=
set GOTOOLDIR=c:\go\pkg\tool\windows_amd64
set GCCGO=gccgo
set AR=ar
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=E:\src\rainway-metapod\go.mod
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=C:\Users\josh\AppData\Local\Temp\go-build763362010=/
tmp/go-build -gno-record-gcc-switches
</pre></details>

### What did you do?
`go get github.com/RainwayApp/metapod@modulize` and then attempt to import and use metapod in a program

```go
package main

import "github.com/RainwayApp/metapod"

func main() {
	metapod.Create()
}
```

### What did you expect to see?
No errors! woo!

### What did you see instead?
```module github.com/RainwayApp/metapod@latest found (v0.0.0-20190515153323-337878358c65), but does not contain package github.com/RainwayApp/metapod```

Whats interesting is that the downloaded module is devoid of go files which I dont really get (this is even after trying to go get the module again after deleting it from here).
![](https://i.f1ssi0n.com/BalneologicGeneticMoa.png)

I got around this by creating a new branch (modulize2) which has all the files - however im not sure how its possible for it to end up in this state in the first place so just thought id bring it up...

**Comment by seankhliao** [comment]
The code is in gitgub.com/RainwayApp/MetaPod/src/metapod

**Comment by bcmills** [comment]
Duplicate of #31866

**Comment by bcmills** [comment]
Note that commit `337878358c65` of that repository indeed does not contain any Go source files at the module root.

**Comment by bcmills** [comment]
`github.com/RainwayApp/metapod@modulize` probably resolved to an older-than-expected commit due to caching in `proxy.golang.org`. This is a known and intentional behavior of the proxy, but it can certainly be confusing.

(I'm trying to find the right issue to mark this as a duplicate.)

**Comment by emily33901** [comment]
I sort of expected that to be the case - just didnt know where in the chain that happened. Is there any reason why the commit hash in the error message is different to the one in the module cache path (`337878358c65` vs `6a6`...)?

**Comment by emily33901** [comment]
Yeah and the error message that i copied in i think is wrong it should be for `@modulize` not `@latest` becuase latest definitely doesnt have any at the root folder

**Comment by icholy** [comment]
Perhaps pkg.go.dev could have a "check for newer version" button. The TTL could be communicated by greying that button out with a "try again in X minutes" message.

**Comment by twmb** [comment]
This also bites repos that use multiple submodules.

```
top
top/pkg/sub
```

If I push tag `v1.17.1` a year ago, and I push `top/pkg/sub@1.17.1`, it takes an unknown amount of time for `top/pkg/sub@1.17.1` to be available to be pulled. As a concrete example, I pushed `pkg/kadm@v1.17.1` 20 minutes ago from right now:

```
$ go get -u github.com/twmb/franz-go/pkg/kadm@v1.17.1
go: module github.com/twmb/franz-go@v1.17.1 found, but does not contain package github.com/twmb/franz-go/pkg/kadm
```

I may have been able to work around it in the last 30s by doing `go get -u ...@<commit sha>`, which then forced a pull from github and discovered the new tag. Or, the remote cache expired. I'll try the commit sha sooner next time I run into this.
